src/interface/src/app/services/plan-notes.service.ts

```diff
@@ -10,43 +10,77 @@ export interface Note {
   created_at: string;
 }
 
+export interface NoteModel {
+  name: string;
+  urlMultiple: (modelObjectId: number) => string;
+  urlSingle: (modelObjectId: number, noteId: number) => string;
+}
+
+const noteModels: NoteModel[] = [
+  {
+    name: 'planning_area',
+    urlMultiple: (planningAreaId: number) =>
+      `/planning/planning_area/${planningAreaId}/note`,
+    urlSingle: (planningAreaId: number, noteId: number) =>
+      `/planning/planning_area/${planningAreaId}/note/${noteId}`,
+  },
+];
+
+function getMultipleUrl(
+  modelName: string,
+  modelObjectId: number
+): string | undefined {
+  const model = noteModels.find((m) => m.name === modelName);
+  return model ? model.urlMultiple(modelObjectId) : undefined;
+}
+
+function getSingleUrl(
+  modelName: string,
+  modelObjectId: number,
+  noteId: number
+): string | undefined {
+  const model = noteModels.find((m) => m.name === modelName);
+  return model ? model.urlSingle(modelObjectId, noteId) : undefined;
+}
+
 @Injectable({
   providedIn: 'root',
 })
-export class PlanNotesService {
+export class ModelNotesService {
   constructor(private http: HttpClient) {}
+  model: string = 'planning_area';
```

Comment:
I guess this is an intermediate step and this will later by more dynamic?

Reply:
it _was_ an intermediate step...removed